provider_client.go

```diff
@@ -88,18 +88,19 @@ type RequestOpts struct {
 	// RawBody contains an io.ReadSeeker that will be consumed by the request directly. No content-type
 	// will be set unless one is provided explicitly by MoreHeaders.
 	RawBody io.ReadSeeker
-
 	// JSONResponse, if provided, will be populated with the contents of the response body parsed as
 	// JSON.
 	JSONResponse interface{}
 	// OkCodes contains a list of numeric HTTP status codes that should be interpreted as success. If
 	// the response has a different code, an error will be returned.
 	OkCodes []int
-
 	// MoreHeaders specifies additional HTTP headers to be provide on the request. If a header is
 	// provided with a blank value (""), that header will be *omitted* instead: use this to suppress
 	// the default Accept header or an inferred Content-Type, for example.
 	MoreHeaders map[string]string
+	// ErrorType specifies the resource error type to return if an error is encountered.
+	// This lets resources override default error messages based on the response status code.
+	ErrorType error
```

Comment:
Another naming nit: it's not really a "type" here, since you're passing a preconstructed error template of sorts. `ErrorContext` maybe, because it's really used to capture request values that can be used to build the eventual error? `ErrorTemplate`? `ErrorExemplar`?